frontend-cert/js-projects/second-oop/user-stories.md

```diff
@@ -0,0 +1,38 @@
+1. You should define a class named `BankAccount` with a `constructor` that has a `balance` parameter (default value of 0). The constructor should also initialize an empty `transactions` array to store transaction records.
+
+2. You should define a method named `deposit` with an `amount` parameter. This method should:
+    - Check if the `amount` is greater than 0.
+    - If the amount is valid, push an object with the properties `type: 'deposit'` and the `amount` into the `transactions ` array.
+    - Call the `updateBalance` method to update the account's balance.
+    - Log a message to the console indicating the deposit was successful and display the new balance.
+    - If the `amount` is invalid, log an error message indicating that the deposit amount must be greater than zero.
```

Comment:
this is too step by step and doesn't allow the camper to think through what needs to be done.
this should be more generalized so the camper is forced to think about how to solve the problem

Reply:
Also, it will be much easier from a testing perspective to return the string instead of logging it to the console.